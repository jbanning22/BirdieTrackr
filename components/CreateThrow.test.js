import React from "react";
import {fireEvent, render, screen} from "@testing-library/react-native";
import CreateThrowScreen from "./CreateThrow.jsx";
import AsyncStorage from "@react-native-async-storage/async-storage";
import axios from "axios";

jest.mock("axios");
jest.mock("@react-native-async-storage/async-storage");

describe("Create Throw Screen", () => {
	beforeEach(() => {
		AsyncStorage.getItem.mockResolvedValue("dummy_token");
		axios.get.mockResolvedValue({
			data: {
				id: 45,
				createdAt: "2023-05-24T14:25:42.716Z",
				disc: "boss",
				color: null,
				throwtype: "forehand",
				distance: "450",
				userId: 1,
			},
		});
	});
	afterEach(() => {
		jest.clearAllMocks();
	});
	it("should match the snapshot", () => {
		const route = {
			params: {dist: 100},
		};
		const {toJSON} = render(<CreateThrowScreen route={route} />);
		expect(toJSON()).toMatchSnapshot();
	});
	it("should render title", () => {
		const route = {
			params: {dist: 100},
		};
		render(<CreateThrowScreen route={route} />);
		expect(screen.getByText("Measure Throw")).toBeTruthy();
	});
	it("should display create throw button", () => {
		const route = {
			params: {dist: 100},
		};
		render(<CreateThrowScreen route={route} />);
		expect(screen.getByText("Create Throw")).toBeTruthy();
	});
	it("should navigate back to throws screen upon throw creation", async () => {
		const route = {
			params: {dist: 100},
		};
		const navigation = {
			navigate: jest.fn(),
		};
		const {getByText} = render(
			<CreateThrowScreen route={route} navigation={navigation} />,
		);
		await fireEvent.press(getByText("Create Throw"));
		expect(navigation.navigate).toHaveBeenCalledWith("ThrowsScreen");
	});
});
