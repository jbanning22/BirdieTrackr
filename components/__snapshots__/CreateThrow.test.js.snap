// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Throw Screen should match the snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/BasketBackground2.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginLeft": 15,
        "marginTop": 35,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontFamily": "Helvetica",
          "fontSize": 28,
          "fontWeight": "500",
          "marginBottom": 80,
        }
      }
    >
      Measure Throw
    </Text>
    <View
      style={
        {
          "backgroundColor": "#F9FAFB",
          "borderRadius": 15,
          "elevation": 5,
          "marginTop": 2,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <TextInput
        clearButtonMode="always"
        onChangeText={[Function]}
        placeholder="Disc Name"
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 50,
            "marginBottom": 15,
            "padding": 10,
            "width": 200,
          }
        }
        value=""
      />
      <TextInput
        clearButtonMode="always"
        onChangeText={[Function]}
        placeholder="Disc Color"
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 50,
            "marginBottom": 15,
            "padding": 10,
            "width": 200,
          }
        }
        value=""
      />
      <TextInput
        clearButtonMode="always"
        onChangeText={[Function]}
        placeholder="Throw Type"
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 50,
            "marginBottom": 15,
            "padding": 10,
            "width": 200,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 30,
        }
      }
    >
      100
       feet
    </Text>
    <Text
      style={
        {
          "color": "#45B369",
          "fontSize": 18,
          "marginBottom": 30,
          "marginTop": 30,
        }
      }
    >
      Great throw!
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#2D6061",
          "borderRadius": 14,
          "height": 56,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 0.4,
          "width": 327,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Satoshi-Medium",
            "fontSize": 18,
            "fontWeight": "500",
          }
        }
      >
        Create Throw
      </Text>
    </View>
  </View>
</View>
`;
